tests/instruments/test_instrument.py

```diff
@@ -213,6 +215,61 @@ def __init__(self):
     assert fake.x == 5
 
 
+@pytest.mark.parametrize("cast, expected", ((float, 5.5),
+                                            (ureg.Quantity, ureg.Quantity(5.5)),
+                                            (str, "5.5"),
+                                            (lambda v: int(float(v)), 5)
+                                            ))
+def test_measurement_cast(cast, expected):
+    class Fake(Instrument):
+        x = Instrument.measurement(
+            "x", "doc", cast=cast)
+    with expected_protocol(Fake, [("x", "5.5")], name="test") as instr:
+        assert instr.x == expected
+
+
+def test_measurement_cast_int():
+    class Fake(Instrument):
+        def __init__(self, adapter, **kwargs):
+            super().__init__(adapter, "test", **kwargs)
+        x = Instrument.measurement(
+            "x", "doc", cast=int)
+    with expected_protocol(Fake, [("x", "5")]) as instr:
+        y = instr.x
+        assert y == 5
+        assert type(y) is int
+
+
+def test_measurement_unitful_property():
+    class Fake(Instrument):
+        def __init__(self, adapter, **kwargs):
+            super().__init__(adapter, "test", **kwargs)
+        x = Instrument.measurement(
+            "x", "doc", get_process=lambda v: ureg.Quantity(v, ureg.m))
+    with expected_protocol(Fake, [("x", "5.5")]) as instr:
+        y = instr.x
+        assert y.m_as(ureg.m) == 5.5
+
+
+@pytest.mark.parametrize("dynamic", [False, True])
+def test_measurement_dict_str_map(dynamic):
```

Comment:
wait is this test just copied from below?

Reply:
Funny, my local file does not have it twice...

Reply:
Huh, that's weird!

Reply:
Now it's here (and is going to be deleted)